go/libraries/doltcore/sqle/dfunctions/dolt_commit.go

```diff
@@ -81,10 +82,24 @@ func (d DoltCommitFunc) Eval(ctx *sql.Context, row sql.Row) (interface{}, error)
 
 	apr := cli.ParseArgs(ap, args, nil)
 
-	// Check if the -all param is provided. Stage all tables if so.
 	allFlag := apr.Contains(cli.AllFlag)
+	allowEmpty := apr.Contains(cli.AllowEmptyFlag)
```

Comment:
This is substantially duplicating logic from the command / action packages. After you finish fixing RepoStateReader, you should take a pass at eliminating some of this duplication.